src/planning/src/planning/smoothing2.cpp

```diff
@@ -0,0 +1,31 @@
+#include "planning/smoothing2.hpp"
+
+void PathSmoothing::order_path(std::vector<PathPoint>& unord_path, const Pose& car_pose) {
+  std::unordered_set<PathPoint> unord_set(unord_path.begin(), unord_path.end());
+
+  PathPoint current_point = PathPoint(car_pose.position, 1);
+  int index = 0;
```

Comment:
Not for this PR obviously, but it is something to note that this might be quite optimizable later